src/OpenSage.Mathematics/MathUtility.cs

```diff
@@ -43,18 +45,40 @@ public static float GetPitchFromDirection(in Vector3 direction)
     }
 
     /// <summary>
-    /// Calculates delta between the 2 angles (which must be between -180 and 180)
+    /// Calculates delta between the 2 angles. The input angles need not be normalized.
+    /// The result will be normalized.
     /// </summary>
     /// <param name="alpha">the first angle</param>
     /// <param name="beta">the second angle</param>
     /// <returns>the absolute delta between the two angles in radians.
-    /// The value is between -PI and PI degrees by definition.
+    /// The value is between -PI and PI degrees.
     /// </returns>
     public static float CalculateAngleDelta(float alpha, float beta)
     {
-        var delta = beta - alpha;
-        delta += (delta > MathF.PI) ? -TwoPi : (delta < -MathF.PI) ? TwoPi : 0;
-        return delta;
+        return NormalizeAngle(alpha - beta);
+    }
+
+    public static float NormalizeAngle(float angle)
+    {
+        Debug.Assert(!float.IsNaN(angle), "Angle is NaN");
+
+        if (float.IsNaN(angle))
+        {
+            // Argh! Don't assert and then not handle it! Error bad! Fix error!
```

Comment:
Lol. Add a todo so we can find this later?

Reply:
This is actually a comment from the original code. I'm not sure if we should throw an exception here - I guess that's what the comment implies - but I think it's okay as-is, because we'll hit the assert in debug builds.

Reply:
Is this something that's worth logging? Just thinking about a case where we forget about this and then ship a build and them someone reports a weird bug, it might be nice if this at least surfaced in logs so we'd know this happened?

Amusing that this came from the original code lol

Reply:
For now I've thrown an exception - 7bf5891fbe3c4fc69d02f9c8e81361714a299baf

If we want to downgrade it to a logged message in the future, we can do that, but I think for now I'd rather know about these kind of issues in both debug and release builds.